poly/helper_test.go

```diff
@@ -0,0 +1,106 @@
+package poly
+
+import (
+	"fmt"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"github.com/dedis/crypto/random"
+)
+
+var edward = SUITE
+
+func generateKeyPair() *config.KeyPair {
+	keypair := new(config.KeyPair)
+	keypair.Gen(edward, random.Stream)
+	return keypair
+}
+
+func generateKeyPairList(n int) []*config.KeyPair {
+	l := make([]*config.KeyPair, n)
+	for i := 0; i < n; i++ {
+		l[i] = generateKeyPair()
+	}
+	return l
+}
+
+func generatePublicListFromPrivate(private []*config.KeyPair) []abstract.Point {
+	l := make([]abstract.Point, len(private))
+	for i := 0; i < len(private); i++ {
+		l[i] = private[i].Public
+	}
+	return l
+}
+
+// produce M receivers from their private/pub keys
+func generateReceivers(info PolyInfo, keys []*config.KeyPair) []*Receiver {
+	n := len(keys)
+	l := make([]*Receiver, n)
+	for i := 0; i < n; i++ {
+		l[i] = NewReceiver(info, keys[i])
+	}
+	return l
+}
+
+// Produce N dealers with the public keys of the M receivers
+func generateDealers(n int, info PolyInfo, receiverList []abstract.Point) []*Dealer {
+	d := make([]*Dealer, n)
+	for i := 0; i < n; i++ {
+		d[i] = NewDealer(info, generateKeyPair(), generateKeyPair(), receiverList)
+	}
+	return d
+}
+
+// Returns N dealers with M receivers with the right keys / public keys ...
+func generateNDealerMReceiver(info PolyInfo, n, m int) ([]*Dealer, []*Receiver) {
+	receiverKeys := generateKeyPairList(m)
```

Comment:
generateKeyPair() is called in the generateDealers() method. 
It's Ok for generateReceivers / Dealers.
I will not argue more here since this is only for a helper test file but I will still say one last thing ;) From a global point of view, I think having separate functions for separate functionalities is always better, even if you create a function of 4 lines, than having a bigger functions that does many things. (But of course if your goal is to have a minimum l.o.c number, then this is not the thing to do !) 
I've merged the dealers / receivers part in the matrix setup.